### PR TITLE
helprank/data -- updateUserStats + sequentialExec

### PR DESCRIPTION
**Pull request by raymondng**
An experimental sequentialExec. Please review. Thanks!

@yingjieMiao @ymatsuda

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/AttributionCommander.scala [hunk 1]
```diff
@@ -109,15 +109,24 @@ class AttributionCommander @Inject() (
     }
   }
 
-  def updateAllReKeepStats(n:Int = 3): Future[Int] = { // expensive -- admin only
+  def sequentialExec[T](futures:List[() => Future[T]]):Future[List[T]] = { // experimental -- generic (does not require special context) but a bit more work for clients
+    if (futures.isEmpty) Future.successful(List.empty)
```
Comment (yingjieMiao) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/AttributionCommander.scala:113]: space after `:`, thanks

Comment (raymondng) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/AttributionCommander.scala:113]: dude ... style police -- okay thanks :)

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/AttributionCommander.scala [hunk 2]
```diff
@@ -109,15 +109,24 @@ class AttributionCommander @Inject() (
     }
   }
 
-  def updateAllReKeepStats(n:Int = 3): Future[Int] = { // expensive -- admin only
+  def sequentialExec[T](futures:List[() => Future[T]]): Future[List[T]] = { // experimental -- generic (does not require special context) but a bit more work for clients
```
Comment (raymondng) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/AttributionCommander.scala:112]: One thing I have some mixed feelings about is the explicit use of () -- it's not the most elegant way to do it but as discussed with @ymatsuda for now I'll leave it the way it is.

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/AttributionCommander.scala (hunk 1 above)
Comment (2is10) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/AttributionCommander.scala:113]: consistent style is important. we need to make our team’s norms into habits. keep pulling us over, @yingjieMiao!

**Pull request merged**